superset/assets/src/visualizations/nvd3/NVD3Vis.js

```diff
@@ -319,10 +319,22 @@ function nvd3Vis(element, props) {
         chart.lines.dispatch.on('elementClick', function (e) {
           if (tableFilter) {
             const publishedColumns = formData.publishColumns;
-            const yColumn = formData.metrics[e.seriesIndex].column.column_name
+            let yColumn;
+            let metric;
+            if (formData.metrics.length == 1) {
+              metric = formData.metrics[0];
+            } else {
+              let key = e.series.key;
+              metric = formData.metrics.find((metric) => {
+                return key.includes(metric.label) ? true : false;
+              })
+            }
+            if (metric.column) {
+              yColumn = metric.column.column_name;
+            }
             const xField = formData.granularitySqla;
             const yField = findYAxisField(yColumn, publishedColumns);
-            if (xField != undefined && e.point) onAddFilter(xField, e.point.x, false, false);
+            if (xField != undefined && e.point) onAddFilter(xField, e.point.x, false, yField ? false : true);
```

Comment:
(suggestion) Since your inline if is return `false` or `true` and that your condition is purely based on the opposite of the thruthly value of `yField` couldn't you simply replace `yField ? false : true` by `!yField` ?

Reply:
Thanks for notify. We will update.